plugins/fix-china-map-offset.js

```diff
@@ -0,0 +1,313 @@
+    // They return distortions in terms of arc lengths, in meters.
+    var dLat_m = -100 + 2 * x + 3 * y + 0.2 * y * y + 0.1 * x * y + 0.2 * Math.sqrt(Math.abs(x)) +
+      20 / 3 * (
+        2 * Math.sin(x * 6 * Math.PI) + 2 * Math.sin(x * 2 * Math.PI) +
+        2 * Math.sin(y * Math.PI) + 4 * Math.sin(y / 3 * Math.PI) +
+        16 * Math.sin(y / 12 * Math.PI) + 32 * Math.sin(y / 30 * Math.PI));
+    var dLon_m = 300 + x + 2 * y + 0.1 * x * x + 0.1 * x * y + 0.1 * Math.sqrt(Math.abs(x)) +
+      20 / 3 * (
+        2 * Math.sin(x * 6 * Math.PI) + 2 * Math.sin(x * 2 * Math.PI) +
+        2 * Math.sin(x * Math.PI) + 4 * Math.sin(x / 3 * Math.PI) +
+        15 * Math.sin(x / 12 * Math.PI) + 30 * Math.sin(x / 30 * Math.PI));
+
+    var radLat = wgs.lat / 180 * Math.PI;
+    var magic = 1 - GCJ_EE * Math.pow(Math.sin(radLat), 2); // just a common expr
+
+    // Arc lengths per degree, on the wrong ellipsoid
+    var lat_deg_arclen = Math.PI / 180 * (GCJ_A * (1 - GCJ_EE)) * Math.pow(magic, 1.5);
+    var lon_deg_arclen = Math.PI / 180 * (GCJ_A * Math.cos(radLat) / Math.sqrt(magic));
+
+    return {
+      lat: wgs.lat + dLat_m / lat_deg_arclen,
+      lng: wgs.lng + dLon_m / lon_deg_arclen,
+    };
+  };
+
+  return gcjObfs;
+})();
+
+plugin.fixChinaOffset.process = function (wgs) {
+  return plugin.fixChinaOffset.isInGoogle(wgs)
+    ? plugin.fixChinaOffset.gcjObfs(wgs)
+    : wgs;
+};
+
+///////// begin overwrited L.GridLayer /////////
+
+L.GridLayer.prototype._getTiledPixelBounds = (function () {
+  return function (center) {
+  /// modified here ///
+    center = window.plugin.fixChinaOffset.getLatLng(center, this.options);
+    /////////////////////
+    var map = this._map,
+      mapZoom = map._animatingZoom ? Math.max(map._animateToZoom, map.getZoom()) : map.getZoom(),
+      scale = map.getZoomScale(mapZoom, this._tileZoom),
+      pixelCenter = map.project(center, this._tileZoom).floor(),
+      halfSize = map.getSize().divideBy(scale * 2);
+
+    return new L.Bounds(pixelCenter.subtract(halfSize), pixelCenter.add(halfSize));
+  };
+})(L.GridLayer.prototype._getTiledPixelBounds);
+
+L.GridLayer.prototype._setZoomTransform = (function (original) {
+  return function (level, center, zoom) {
+    center = window.plugin.fixChinaOffset.getLatLng(center, this.options);
+    original.apply(this, [level, center, zoom]);
+  };
+})(L.GridLayer.prototype._setZoomTransform);
+
+L.GridLayer.GoogleMutant.prototype._update = (function (original) {
+  return function () {
+    this.options.needFixChinaOffset = true;
+    // zoom level check needs to happen before super's implementation (tile addition/creation)
+    // otherwise tiles may be missed if maxNativeZoom is not yet correctly determined
+    if (this._mutant) {
+      var center = this._map.getCenter();
+      /// modified here ///
+      center = window.plugin.fixChinaOffset.getLatLng(center, this.options);
+      /////////////////////
```

Comment:
In order to make eslint happy
```suggestion
      /////////////////////
      /* eslint-disable */
```